qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,255 @@
+            ansatz: the UCC ansatz.
+
+        Return:
+            The computed initial point.
+        """
+        self.driver_result = driver_result
+        self.ansatz = ansatz
+        self._corrections = self._compute_corrections()
+        return self.initial_point
+
+    def _compute_corrections(
+        self,
+    ) -> dict[str, tuple[float, float]]:
+        """Compute the MP2 coefficients and energy corrections for each double excitation.
+
+        Non-double excitations will have zero coefficient and energy delta.
+
+        Returns:
+            Dictionary with MP2 coefficients and energy_deltas for each excitation
+        """
+        corrections = {}
+        for excitation in self._excitations:
+            if len(excitation[0]) == 2:
+                # MP2 needs double excitations.
+                coeff, e_delta = self._compute_correction(excitation)
+            else:
+                # Leave single excitations unchanged.
+                coeff, e_delta = 0.0, 0.0
+
+            corrections[str(excitation)] = (coeff, e_delta)
+
+        return corrections
+
+    def _compute_correction(
+        self, excitation: tuple[tuple[int, ...], tuple[int, ...]]
+    ) -> tuple[float, float]:
+        """Compute the MP2 coefficient and energy corrections given a double excitation.
+
+        Each double excitation given by [i,a,j,b] has a coefficient computed using
+            coeff = -(2 * T[i,a,j,b] - T[i,b,j,a)/(E[b] + E[a] - E[i] - E[j])
+        where E is the orbital energy.
+        and an energy delta given by
+            e_delta = coeff * T[i,a,j,b]
+
+        All the computations are done using the molecule orbitals but the indices used
+        in the excitation information passed in and out are in the block spin orbital
+        numbering as normally used by the nature module:
+          - alpha runs from 0 to num_orbitals - 1
+          - beta runs from num_orbitals to num_orbitals * 2 - 1
+
+        Args:
+            excitations: Sequence of excitations.
+
+        Returns:
+            Correction coefficients and energy deltas.
+        """
+        # Since spins are the same drop to MO indexing
+        num_orbitals = self._num_spin_orbitals // 2
+        i = excitation[0][0] % num_orbitals
+        j = excitation[0][1] % num_orbitals
+        a = excitation[1][0] % num_orbitals
+        b = excitation[1][1] % num_orbitals
+
+        tiajb = self._integral_matrix[i, a, j, b]
+        tibja = self._integral_matrix[i, b, j, a]
+
+        num = 2 * tiajb - tibja
+        denom = (
+            self._orbital_energies[b]
+            + self._orbital_energies[a]
+            - self._orbital_energies[i]
+            - self._orbital_energies[j]
+        )
+        coeff = -num / denom
+        coeff = coeff if abs(coeff) > self._threshold else 0
+        e_delta = coeff * tiajb
+        e_delta = e_delta if abs(e_delta) > self._threshold else 0
```

Comment:
Can we use better names, e.g. `correction_coeff`, `energy_delta` etc. How about `denom`? Can we assign it some meaning?

Reply:
Sure, I've now made these variable names more explicit.